doc/source/user_guide/documentation/types.rst

```diff
@@ -0,0 +1,11 @@
+Types
+=====
+
+.. automodule:: icepyx.core.types
+   :members:
+   :undoc-members:
+   :exclude-members: CMRParamsBase,CMRParamsWithBbox,CMRParamsWithPolygon
+
+.. COMMENT. `exclude-members` specified above is required because those models
+   contain symbols ('[', ']') in some keys, which sphinx doesn't like.
+   See: https://github.com/sphinx-doc/sphinx/issues/11039
```

Comment:
"[" in a key strikes again!

Reply:
Yes! Perhaps we can address that in v2 by switching from dictionaries to Pydantic objects (which can be read in from dicts or converted to dicts or query strings that look the way we need)!

Reply:
Or the folks on your end could stop creating keys with symbols in them. 🤣 I'm intrigued by these Pydantic objects and how they could help with this challenge.

Reply:
>Or the folks on your end could stop creating keys with symbols in them.

Hah, I don't think I have that much pull ;)

>I'm intrigued by these Pydantic objects and how they could help with this challenge.

Pydantic is super cool! We can use Pydantic models to describe the objects we pass around in the code, and control [serialization](https://docs.pydantic.dev/latest/concepts/serialization/), and also to do automated runtime validation based on type hints!